middlewares/accesslog/linear_writer_test.go

```diff
@@ -0,0 +1,59 @@
+package accesslog
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLinearWriter(t *testing.T) {
+	file := logfilePath("")
+
+	w, err := os.Create(file)
+	assert.Nil(t, err, "%v", err)
+
+	lw := LinearWriter(w)
+	wg := &sync.WaitGroup{}
+
+	n := 10
+	wg.Add(n)
+	for i := 0; i < n; i++ {
+		go func(j int) {
+			fmt.Fprintf(lw, "Line %d\n", j)
+			wg.Done()
+		}(i)
+	}
+	wg.Wait()
+
+	err = lw.Close()
+	assert.Nil(t, err, "%v", err)
+
+	// read back and verify
+	counts := make([]int, n)
+	r, err := os.Open(file)
+	assert.Nil(t, err, "%v", err)
+	defer r.Close()
+	defer os.Remove(file)
+
+	scanner := bufio.NewScanner(r)
+	for i := 0; i < n; i++ {
+		assert.True(t, scanner.Scan())
+		line := scanner.Text()
```

Comment:
I think this implementation is quite simple.  I had a quick try with ReadFile but it seemed more complicated.